src/js/pages/Hukamnama/Layout.js

```diff
@@ -1,36 +1,86 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { TEXTS } from '../../constants';
+import { Link } from 'react-router-dom';
+import { TEXTS, FIRST_HUKAMNAMA_DATE } from '../../constants';
 import { pageView } from '../../util/analytics';
 import ShabadContent from '../../components/ShabadContent';
 import BreadCrumb from '../../components/Breadcrumb';
+import { dateMath } from '../../util/index';
+import GenericError, { BalpreetSingh } from '../../components/GenericError';
 
 export const Stub = () => <div className="spinner" />;
 
 export default class Layout extends React.PureComponent {
   static propTypes = {
     data: PropTypes.object.isRequired,
   };
+
+  state = {
+    error: false,
```

Comment:
```suggestion
    error: this.props.data.error,
```

No need of constructor